tests/end-to-end/external-solution-integration/pkg/step/step.go

```diff
@@ -100,3 +112,10 @@ func isNotFound(err error) bool {
 	})
 	return isNotFound
 }
+
+type CleanupBehavior string
+
+const (
+	CleanupBehaviorAllSteps         CleanupBehavior = "all"
```

Comment:
What is the reason behind introducing this? Why is a cleanup of all steps required even though the step did not get a change to create resources? 🤔